node/src/chain.rs

```diff
@@ -528,20 +366,272 @@ pub async fn create_ethereum_networks_for_chain(
             ),
             web3.limit_for(&config.node),
         );
+
+        if call_only {
+            call_only_adapters.push(adapter);
+        } else {
+            adapters.push(adapter);
+        }
     }
 
-    parsed_networks.sort();
-    Ok(parsed_networks)
+    adapters.sort_by(|a, b| {
+        a.capabilities
+            .partial_cmp(&b.capabilities)
+            // We can't define a total ordering over node capabilities,
+            // so incomparable items are considered equal and end up
+            // near each other.
+            .unwrap_or(Ordering::Equal)
+    });
+
+    Ok(AdapterConfiguration::Rpc(EthAdapterConfig {
+        chain_id: network_name.into(),
+        adapters,
+        call_only: call_only_adapters,
+        polling_interval: Some(chain.polling_interval),
+    }))
```

Comment:
Why doesn't `EthAdapterConfig` have a constructor that sorts the adapters? I assume the ordering is important for the inner workings of it, and that would be a good place to do the sorting.